### ISSUE TITLE
SoD phase 5 bug, WA window wont show "Add New" or "search" UI feature and when I open WAs it generates an error. 

### ISSUE THREAD

**Issue by Abrams011** [WeakAuras/WeakAuras2#5462]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues.

### Description

Since the last patch every version of WAs has been missing the "Add New" and "search" feature in the WA UI window. Currently using "5.17.1-34-g514f3b4" and the bug still persists. I disabled all addons except WAs and buggrabber/bugsack and I get a "green" no bugs result until I open WAs where then it generates the error in the attached photo. 
![WoWScrnShot_100524_090310](https://github.com/user-attachments/assets/e8bf91b9-4f7a-4a05-8cba-e633ee859986)
![WoWScrnShot_100524_090013](https://github.com/user-attachments/assets/0c950acc-d346-46c1-ab2e-183b7a5a475b)

### WeakAuras Version

5.17.1-34-g514f3b4

### World of Warcraft Flavor

Classic Era

### World of Warcraft Region

US/NA

### Tested with only WeakAuras

I got this issue with only WeakAuras enabled

### Lua Error

```Text
1x WeakAuras/GenericTrigger.lua:3451: Usage: FindSpellOverrideByID(spellID)
[string "=[C]"]: ?
[string "@WeakAuras/GenericTrigger.lua"]:3451: in function `GetEffectiveSpellId'
[string "@WeakAuras/Prototypes.lua"]:2247: in function `GetNameAndIcon'
[string "@WeakAuras/GenericTrigger.lua"]:1064: in function <WeakAuras/GenericTrigger.lua:1050>
[string "@WeakAuras/GenericTrigger.lua"]:1100: in function `CreateFakeStates'
[string "@WeakAuras/WeakAuras.lua"]:4598: in function `UpdateFakeStatesFor'
[string "@WeakAuras/WeakAuras.lua"]:4574: in function `FakeStatesFor'
[string "@WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua"]:1545: in function `SyncVisibility'
[string "@WeakAurasOptions/WeakAurasOptions.lua"]:871: in function `ShowOptions'
[string "@WeakAurasOptions/WeakAurasOptions.lua"]:613: in function `ToggleOptions'
[string "@WeakAuras/WeakAuras.lua"]:172: in function `OpenOptions'
[string "@WeakAuras/WeakAuras.lua"]:1149: in function `OnClick'
[string "@BugSack/Libs/LibDBIcon-1.0-55/LibDBIcon-1.0.lua"]:179: in function <BugSack/Libs/LibDBIcon-1.0/LibDBIcon-1.0.lua:177>

Locals:
(*temporary) = "Victory Rush"
```

### Reproduction Steps

Click on WAs options to load the options UI to create or edit WAs and the error happens. 

### Screenshots

![WoWScrnShot_100524_090310](https://github.com/user-attachments/assets/99940b7b-0b36-49c8-84b9-18d02d7d4655)
![WoWScrnShot_100524_090013](https://github.com/user-attachments/assets/b79f9176-4714-49bc-908f-4df6cd6a0265)

**Comment by InfusOnWoW** [comment]
We need your settings to reproduce that.

**Comment by Abrams011** [comment]
> We need your settings to reproduce that.

Which settings specifically. WeakAurasOptions?

**Comment by InfusOnWoW** [comment]
No, the WeakAuras.lua file, the initial bug template asked you to include.

**Comment by SpatenLa** [comment]
they put it at the very top in a later edit

**Comment by Abrams011** [comment]
> No, the WeakAuras.lua file, the initial bug template asked you to include

I added a .zip file. Looks like it appears at the top of the post on my screen.

**Comment by mrbuds** [comment]
I dont have a warrior, but i can reproduce by clicking the visibility button for unloaded auras

Error happen when Private.ExecEnv.GetEffectiveSpellId which call GetOverrideSpell which is FindSpellOverrideByID, is called with a string "Victory Rush", instead of a number

data from trigger making error

```Lua
  ["trigger"] = {
    ["duration"] = "0",
    ["genericShowOn"] = "showOnCooldown",
    ["unit"] = "player",
    ["use_direction"] = true,
    ["debuffType"] = "HELPFUL",
    ["use_sourceName"] = false,
    ["subeventSuffix"] = "_CAST_SUCCESS",
    ["use_genericShowOn"] = true,
    ["event"] = "Action Usable",
    ["use_sourceFlags2"] = false,
    ["realSpellName"] = "Victory Rush",
    ["use_spellName"] = true,
    ["sourceUnit"] = "player",
    ["use_sourceUnit"] = true,
    ["type"] = "spell",
    ["subeventPrefix"] = "SPELL",
    ["use_track"] = true,
    ["spellName"] = "Victory Rush"
  }
```

**Comment by mrbuds** [comment]
AddFakeInformation
=> GetNameAndIconForSpellName(trigger)
=> Private.ExecEnv.GetEffectiveSpellId(trigger.spellName, trigger.use_exact_spellName, not trigger.use_ignoreoverride)
=> FindSpellOverrideByID("Victory Rush")

**Comment by InfusOnWoW** [comment]
Yeah I have a fix, I was assuming only numbers in the spell field, which turns out to be wrong in pratice.
